chapter_preliminaries/linear-algebra.ipynb

```diff
@@ -0,0 +1,1881 @@
+    "lower-cased letters (e.g., $\\mathbf{x}$, $\\mathbf{y}$, and $\\mathbf{z})$.\n",
+    "\n",
+    "We work with vectors via one-dimensional NDArrays.\n",
+    "In general NDArrays can have arbitrary lengths,\n",
+    "subject to the memory limits of your machine.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "ND: (4) cpu() float32\n",
+       "[0., 1., 2., 3.]\n"
+      ]
+     },
+     "execution_count": 7,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "NDArray x = nd.arange(4f);\n",
+    "x"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "origin_pos": 8
+   },
+   "source": [
+    "We can refer to any element of a vector by using a subscript.\n",
+    "For example, we can refer to the $i^\\mathrm{th}$ element of $\\mathbf{x}$ by $x_i$.\n",
+    "Note that the element $x_i$ is a scalar,\n",
+    "so we do not bold-face the font when referring to it.\n",
+    "Extensive literature considers column vectors to be the default\n",
+    "orientation of vectors, so does this book.\n",
+    "In math, a vector $\\mathbf{x}$ can be written as\n",
+    "\n",
+    "$$\\mathbf{x} =\\begin{bmatrix}x_{1}  \\\\x_{2}  \\\\ \\vdots  \\\\x_{n}\\end{bmatrix},$$\n",
+    ":eqlabel:`eq_vec_def`\n",
+    "\n",
+    "\n",
+    "where $x_1, \\ldots, x_n$ are elements of the vector.\n",
+    "In code, we access any element by indexing into the NDArray.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "ND: () cpu() float32\n",
+       "3.\n"
+      ]
+     },
+     "execution_count": 8,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "x.get(3)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "origin_pos": 12
+   },
+   "source": [
+    "### Length, Dimensionality, and Shape\n",
+    "\n",
+    "Let us revisit some concepts from :numref:`sec_ndarray`.\n",
+    "A vector is just an array of numbers.\n",
+    "And just as every array has a length, so does every vector.\n",
+    "In math notation, if we want to say that a vector $\\mathbf{x}$\n",
+    "consists of $n$ real-valued scalars,\n",
+    "we can express this as $\\mathbf{x} \\in \\mathbb{R}^n$.\n",
+    "The length of a vector is commonly called the *dimension* of the vector.\n",
+    "\n",
+    "As with an ordinary Java array, we can access the length. In the case of a NDArray we can achieve this by using the `size()` function."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "4"
+      ]
+     },
+     "execution_count": 9,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "x.size()"
```

Comment:
size() the number of elements in this NDAArray. len() in Python should be the first dimension of this NDArray
```suggestion
    "x.size(0)"
```